test-integration/repay-loan/1_repay_loan_3_payments.js

```diff
@@ -0,0 +1,190 @@
+  // Advance time to make first payment 10 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const paymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(paymentTimestamp);
+
+  // Make 1st payment
+  console.log(`Repaying loan id ${lastLoanID}...`);
+  console.log('Making 1st payment...');
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  totalOwedResult = totalOwedResult.minus(payment);
+  const repay1_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+  loans
+    .loanRepaid(repay1_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+
+  // Check loan status
+  const firstLoanStatus = await loansInstance.loans(lastLoanID);
+  const firstLoanStatusResult = firstLoanStatus.status;
+  assert.equal(
+    firstLoanStatusResult.toString(),
+    BigNumber((2).toString()),
+    'Invalid final loan staus.'
+  );
+
+  // Advance time to make first payment 20 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const secondPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(secondPaymentTimestamp);
+
+  // Make 2nd payment
+  console.log('Making 2nd payment...');
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  totalOwedResult = totalOwedResult.minus(payment);
+  const repay2_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+  // Check loan status
+  loans
+    .loanRepaid(repay2_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const secondLoanStatus = await loansInstance.loans(lastLoanID) 
+  assert.equal(
+    secondLoanStatus.status.toString(),
+    loanStatuses.Active,
+    'Invalid #2 loan staus.'
+  );
+
+  // Advance time to make first payment 29 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const thirdPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(9));
+  await timer.advanceBlockAtTime(thirdPaymentTimestamp);
+
+  // Make 3rd payment
+  console.log('Making final payment...');
+  await token.approve(lendingPoolInstance.address, payment, borrowerTxConfig);
+  const repay3_result = await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+
+  // Check final loan status 
+  totalOwedResult = totalOwedResult.minus(payment);
+  console.log('Expected (after 3# repay) total owed:  ', totalOwedResult.toString());
+  loans
+    .loanRepaid(repay3_result)
+    .emitted(lastLoanID, borrowerTxConfig.from, payment, borrowerTxConfig.from, totalOwedResult.toString());
+  const loanStatus = await loansInstance.loans(lastLoanID);
```

Comment:
Please, add assertions for the loanStatus.principalOwed == 0 and loanStatus.interestOwed == 0

Reply:
Added